File: app/src/main/java/droiddevelopers254/droidconke/ui/BottomNavigationBehaviour.java
package droiddevelopers254.droidconke.ui;

import androidx.annotation.NonNull;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.core.view.ViewCompat;
import android.view.View;

public class BottomNavigationBehaviour extends CoordinatorLayout.Behavior<BottomNavigationView> {
    private int height;

    @Override
    public boolean onLayoutChild(@NonNull CoordinatorLayout parent, @NonNull BottomNavigationView child, int layoutDirection) {
        height= child.getHeight();
        return super.onLayoutChild(parent, child, layoutDirection);
    }

    @Override
    public boolean onStartNestedScroll(@NonNull CoordinatorLayout coordinatorLayout, @NonNull BottomNavigationView child, @NonNull View directTargetChild, @NonNull View target, int axes, int type) {
        return axes == ViewCompat.SCROLL_AXIS_VERTICAL;
    }

    @Override
    public void onNestedScroll(@NonNull CoordinatorLayout coordinatorLayout, @NonNull BottomNavigationView child, @NonNull View target, int dxConsumed, int dyConsumed, int dxUnconsumed, int dyUnconsumed, int type) {
        //super.onNestedScroll(coordinatorLayout, child, target, dxConsumed, dyConsumed, dxUnconsumed, dyUnconsumed, type);
        if (dyConsumed >0){
            slideDown(child);
        }else if (dyConsumed < 0){
            slideUp(child);
        }

    }

    private void slideUp(BottomNavigationView child) {
        child.clearAnimation();
        child.animate().translationY(0).setDuration(200);
    }

    private void slideDown(BottomNavigationView child) {
        child.clearAnimation();
        child.animate().translationY(height).setDuration(200);
    }
}

File: app/src/main/java/droiddevelopers254/droidconke/utils/BottomNavigationBehaviour.kt
package droiddevelopers254.droidconke.utils

import android.view.View
import androidx.coordinatorlayout.widget.CoordinatorLayout
import androidx.core.view.ViewCompat
import com.google.android.material.bottomnavigation.BottomNavigationView

class BottomNavigationBehaviour : CoordinatorLayout.Behavior<BottomNavigationView>() {
  private var height: Int = 0

  override fun onLayoutChild(parent: CoordinatorLayout, child: BottomNavigationView, layoutDirection: Int): Boolean {
    height = child.height
    return super.onLayoutChild(parent, child, layoutDirection)
  }

  override fun onStartNestedScroll(coordinatorLayout: CoordinatorLayout, child: BottomNavigationView, directTargetChild: View, target: View, axes: Int, type: Int): Boolean {
    return axes == ViewCompat.SCROLL_AXIS_VERTICAL
  }

  override fun onNestedScroll(coordinatorLayout: CoordinatorLayout, child: BottomNavigationView, target: View, dxConsumed: Int, dyConsumed: Int, dxUnconsumed: Int, dyUnconsumed: Int, type: Int) {
    //super.onNestedScroll(coordinatorLayout, child, target, dxConsumed, dyConsumed, dxUnconsumed, dyUnconsumed, type);
    if (dyConsumed > 0) {
      slideDown(child)
    } else if (dyConsumed < 0) {
      slideUp(child)
    }

  }

  private fun slideUp(child: BottomNavigationView) {
    child.clearAnimation()
    child.animate().translationY(0f).duration = 200
  }

  private fun slideDown(child: BottomNavigationView) {
    child.clearAnimation()
    child.animate().translationY(height.toFloat()).duration = 200
  }
}
